### ISSUE TITLE
specifying batch size in image_data_param not working

### ISSUE THREAD

**Issue by mfernezir** [NVIDIA/DIGITS#712]
I tried using image_data_param instead of data_param to use shuffle: true option. While doing that, I noticed that specifying batch_size there apparently has no effect. That is, even with provisionally high batch sizes ( even if larger than the number of images in the dataset), GPU memory utilization stays low. I'm guessing some default low batch is being used instead, no matter what is actually specified in the image_data_param. While I was testing this, I removed data_param and didn't specify training and validation batch sizes anywhere else.

**Comment by lukeyeager** [comment]
Which version of DIGITS are you using? In the latest version, all Data layer types other than `Data` and `HDF5Data` are disallowed:

https://github.com/NVIDIA/DIGITS/blob/v3.3.0/digits/model/tasks/caffe_train.py#L1421-L1466

**Comment by mfernezir** [comment]
I am using the Data layer. This is about using different messages in the Data layer. If I specify batch size in the ImageDataParam message rather than in the usual DataParam message, then there is an issue.

Admittedly, this is on 3.2. I can check later today for 3.3.

**Comment by lukeyeager** [comment]
The `image_data_param` field is used to supply options to `ImageData` layers. Use the `data_param` field to supply options to `Data` layers.

https://github.com/NVIDIA/caffe/blob/v0.14.2/src/caffe/proto/caffe.proto#L351-L388

**Comment by mfernezir** [comment]
Ok, I wasn't aware of that. This goes to the list of my Caffe documentation griefs, but at least I've learned something. There is no layer list in caffe.proto and which messages correspond to which layers and doxygen only says "TODO(dox): thorough documentation for Forward and proto params." On the bright side, for the first time I've noticed that in the part you highlighted the first part of the parameter name must correspond to the accompanying layer name.

What's left to complain about though is that Caffe's DataLayer actually silently accepts the ImageDataParam message with shuffle: true option as if that message is actually doing something for that layer type.
